carto/sql.py

```diff
@@ -190,6 +195,37 @@ def create(self, sql_query):
                          http_header=header)
         return data
 
+    def create_and_wait_for_completion(self, sql_query):
+        """
+        Creates a new batch SQL query and waits for its completion or failure
+
+        Batch SQL jobs are asynchronous, once created this method
+        automatically queries the job status until it's one of 'done',
+        'failed', 'canceled', 'unknown'
+
+        :param sql_query: The SQL query to be used
+        :type sql_query: str or list of str
+
+        :return: Response data, either as json or as a regular response.content
+                    object
+        :rtype: object
+
+        :raise: CartoException
+        """
+        header = {'content-type': 'application/json'}
+        data = self.send(self.api_url,
+                         http_method="POST",
+                         json_body={"query": sql_query},
+                         http_header=header)
+
+        warnings.warn('Batch SQL job created with job_id: {job_id}'.format(job_id=data['job_id']))
+
+        while data and data['status'] in BATCH_JOBS_PENDING_STATUSES:
+            time.sleep(BATCH_READ_STATUS_AFTER_SECONDS)
+            data = self.read(data['job_id'])
+
+        return data
```

Comment:
For this method, I'd expect for it to return the data, or an exception if the job failed/was canceled. This is just an opinion and I don't know the SDK too much, the important part is consistency. But in general, in Python, I expect errors to raise an exception rather to rely on the caller checking the return status.